src/common/database.c

```diff
@@ -2729,17 +2746,24 @@ int _get_pragma_val(const struct dt_database_t *db, const char* pragma)
 void dt_database_maybe_maintenance(const struct dt_database_t *db, const gboolean has_gui, const gboolean closing_time)
 {
   gboolean check_for_maintenance = FALSE;
+  gboolean force_maintenance = FALSE;
 
   char *config = dt_conf_get_string("database/maintenance_check");
 
   if(config)
   {
     if((!g_strcmp0(config, "on both"))
-        || (closing_time && !g_strcmp0(config, "on close"))
-        || (!closing_time && !g_strcmp0(config, "on startup")))
+        || (!g_strcmp0(config, "on both (don't ask)"))
```

Comment:
you may want to use g_strstr() on "on both" only. This will divide by 2 the number of conditions.

Reply:
Huh, TIL.